### PR TITLE
feat: Add ping confirmation modal (WPB-3202)

### PR DESCRIPTION
**Pull request by thisisamir98**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-3202" title="WPB-3202" target="_blank">EpicWPB-3202</a>  Ping Confirmation
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

_Briefly describe the issue you have solved or implemented with this pull request. If the PR contains multiple issues, use a bullet list._

### Causes (Optional)

_Briefly describe the causes behind the issues. This could be helpful to understand the adopted solutions behind some nasty bugs or complex issues._

### Solutions

_Briefly describe the solutions you have implemented for the issues explained above._

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by tlebon** [comment]
i guess this is waiting for Design, product, etc. if there is a ticket in jira please link it to this.

**Comment by thisisamir98** [comment]
we have a ticket now: WPB-3202

File: src/script/components/InputBar/InputBar.tsx [hunk 1]
```diff
@@ -565,13 +566,33 @@ const InputBar = ({
...
+
   const onPingClick = () => {
     if (conversationEntity && !pingDisabled) {
-      setIsPingDisabled(true);
-
-      messageRepository.sendPing(conversationEntity).then(() => {
-        window.setTimeout(() => setIsPingDisabled(false), CONFIG.PING_TIMEOUT);
-      });
+      if (conversationEntity.is1to1() || totalConversationUsers < 3) {
+        setIsPingDisabled(true);
```
Comment (PatrykBuniX) [inline src/script/components/InputBar/InputBar.tsx]: Seems like `setIsPingDisabled(true);` could be part of `ping` function, no?

File: src/script/components/InputBar/InputBar.tsx [hunk 2]
```diff
@@ -565,13 +566,33 @@ const InputBar = ({
...
+        ping();
+      } else {
+        PrimaryModal.show(PrimaryModal.type.CONFIRM, {
+          primaryAction: {
+            action: () => {
+              setIsPingDisabled(true);
+              ping();
+            },
```
Comment (PatrykBuniX) [inline src/script/components/InputBar/InputBar.tsx]: After `setIsPingDisabled(true);` is moved to ping, we can simplify this part to just:
```ts
{
  action: ping,
}
```

File: src/script/components/InputBar/InputBar.tsx [hunk 3]
```diff
@@ -565,13 +566,33 @@ const InputBar = ({
...
+  const totalConversationUsers = conversationEntity.participating_user_ets().length;
+
   const onPingClick = () => {
     if (conversationEntity && !pingDisabled) {
-      setIsPingDisabled(true);
-
-      messageRepository.sendPing(conversationEntity).then(() => {
-        window.setTimeout(() => setIsPingDisabled(false), CONFIG.PING_TIMEOUT);
-      });
+      if (conversationEntity.is1to1() || totalConversationUsers < 3) {
```
Comment (PatrykBuniX) [inline src/script/components/InputBar/InputBar.tsx]: What about having a constant for this `"3"`? e.g. `MAX_USERS_TO_PING_WITHOUT_ALERT`?

**Review by arjita-mitra — APPROVED** [review]

File: src/script/components/InputBar/InputBar.tsx [hunk 4]
```diff
@@ -565,13 +567,30 @@ const InputBar = ({
 
   const onGifClick = () => openGiphy(inputValue);
 
+  const ping = () => {
```
Comment (przemvs) [inline src/script/components/InputBar/InputBar.tsx]: I would go with pingConversation or smth like that :)

File: src/script/components/InputBar/InputBar.tsx [hunk 5]
```diff
@@ -565,13 +567,30 @@ const InputBar = ({
...
+    setIsPingDisabled(true);
+    void messageRepository.sendPing(conversationEntity).then(() => {
+      window.setTimeout(() => setIsPingDisabled(false), CONFIG.PING_TIMEOUT);
+    });
+  };
+
+  const totalConversationUsers = conversationEntity.participating_user_ets().length;
```
Comment (przemvs) [inline src/script/components/InputBar/InputBar.tsx]: Use participatingUserEts.length instead :)

File: src/script/components/InputBar/InputBar.tsx [hunk 6]
```diff
@@ -565,13 +567,30 @@ const InputBar = ({
...
+
   const onPingClick = () => {
     if (conversationEntity && !pingDisabled) {
-      setIsPingDisabled(true);
-
-      messageRepository.sendPing(conversationEntity).then(() => {
-        window.setTimeout(() => setIsPingDisabled(false), CONFIG.PING_TIMEOUT);
-      });
+      if (conversationEntity.is1to1() || totalConversationUsers < MAX_USERS_TO_PING_WITHOUT_ALERT) {
```
Comment (przemvs) [inline src/script/components/InputBar/InputBar.tsx]: Get is1to1() from useKoSubscribableChildren :)

File: src/script/components/InputBar/InputBar.tsx [hunk 7]
```diff
@@ -101,6 +102,7 @@ interface InputBarProps {
 }
 
 const conversationInputBarClassName = 'conversation-input-bar';
+const MAX_USERS_TO_PING_WITHOUT_ALERT = 3;
```
Comment (przemvs) [inline src/script/components/InputBar/InputBar.tsx]: Maybe add this to the config ? 🤔

**Review by PatrykBuniX — APPROVED** [review]

File: src/script/components/InputBar/InputBar.tsx [hunk 8]
```diff
@@ -565,13 +567,34 @@ const InputBar = ({
...
+
   const onPingClick = () => {
     if (conversationEntity && !pingDisabled) {
-      setIsPingDisabled(true);
-
-      messageRepository.sendPing(conversationEntity).then(() => {
-        window.setTimeout(() => setIsPingDisabled(false), CONFIG.PING_TIMEOUT);
-      });
+      if (
+        CONFIG.FEATURE.ENABLE_PING_CONFIRMATION === false ||
```
Comment (przemvs) [inline src/script/components/InputBar/InputBar.tsx]: You need to check if === false? :D

Comment (thisisamir98) [reply src/script/components/InputBar/InputBar.tsx]: !CONFIG.FEATURE.ENABLE_PING_CONFIRMATION looks less readable to me

**Review by przemvs — APPROVED** [review]

**Pull request merged**